indico/modules/events/registration/fields/simple.py

```diff
@@ -39,6 +39,28 @@
 from MaKaC.webinterface.common.countries import CountryHolder
 
 
+def get_field_merged_options(field, registration_data):
+    rdata = registration_data.get(field.id)
+
+    try:
+        val = rdata.data.keys()[0]
+    except (AttributeError, IndexError):
+        val = None
+
+    result = field.view_data
```

Comment:
I'd `deepcopy()` it. Chances are good that appending to `choices` modifies a dict that's actually in the database, and the only reason why it doesn't end up committed is that the field is not flagged as modified.